be/src/util/metrics.h

```diff
@@ -161,12 +161,14 @@ class LockSimpleMetric : public Metric {
     T _value;
 };
 
-template <typename T>
-class CoreLocalCounter : public Metric {
+template <typename T, MetricType Type>
+class CoreLocalMetric : public Metric {
 public:
-    CoreLocalCounter(MetricUnit unit) : Metric(MetricType::COUNTER, unit), _value() {}
+    static_assert(Type == MetricType::COUNTER || Type == MetricType::GAUGE,
```

Comment:
this looks a little weird, maybe type traits can be played here to make the template instantiation failure if the Type is not COUNTER and not GAUGE?

Reply:
yeah, both ways are fine.